tests/test_conflicting_formatters.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+from collections.abc import Generator
+from pathlib import Path
+
+import pytest
+from _pytest.fixtures import SubRequest
+
+from pydocstringformatter import _formatting
+from pydocstringformatter import _testutils as test_utils
+from pydocstringformatter._utils import UnstableResultError
+from pydocstringformatter.run import _Run
+
+
+@pytest.fixture(scope="function")
+def patched_run(request: SubRequest) -> Generator[type[_Run], None, None]:
+    """Patches the _Run class to use the MakeAFormatter and MakeBFormatter."""
+    # This monkeypatches the formatters to make them conflict.
+    # and the only formatters applied
+
+    old_formatters = _formatting.FORMATTERS
+    _formatting.FORMATTERS = request.param
+
+    def cleanup() -> None:
+        """Returns the formatters to their original state."""
+        _formatting.FORMATTERS = old_formatters
+
+    yield _Run
+    cleanup()
+
+
+@pytest.mark.parametrize(
+    "patched_run,expect_errors",
+    [
+        (
+            [test_utils.MakeAFormatter(), test_utils.MakeBFormatter()],
+            ["Conflicting Formatters"],
+        ),
+        (
+            [test_utils.MakeBFormatter(), test_utils.AddBFormatter()],
+            ["not able to make stable changes"],
+        ),
+        (
+            [
+                test_utils.MakeBFormatter(),
+                test_utils.AddBFormatter(),
+                test_utils.MakeAFormatter(),
+            ],
+            ["Conflicting formatters:", "Diff too intricate to compute"],
+        ),
+    ],
+    indirect=["patched_run"],
+)
+def test_conflicting_formatters(
+    patched_run: type[_Run],  # pylint: disable=redefined-outer-name
```

Comment:
I just noticed, the typing seems incorrect here. Shouldn't this be `list[Formatters]`?

Reply:
nope, it is actually the _Run class